### PR TITLE
chore: create a comprehensive contribution guideline for Abbreve.

### LINKED ISSUE #4: [DOCS] Add CONTRIBUTING.md file
### Description

To help new contributors know exactly what to do and what to expect of the project...
 - [ ] We might need guidelines on how to fork, clone, push, etc
 - [ ] What kind of contributions we are looking for
 - [ ] Commit conventions, issue, and pull request templates
 - [ ] Links to resources, the code of conduct, etc

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by slim-codes**
* This document specifies the rules and requirements for making contributions to this project.

* Along with step-by-step instructions for **setting up the project** and **making contributions** (Issues, PR's etc)

* These instructions are tailored for potential contributors with little or no experience in making OS contributions. 
Therefore, the instructions may be a bit verbose but comprehensive.

## Fixes Issue

Closes #4 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers
@Njong392 
- I have highlighted 2 sections which require your attention.
- Firstly, there are 2 instances where a link is required for reaching out to maintainers or joining the community if there's one.
I was thinking of using the [Discord link to EddieHub](https://discord.gg/s2PQs2mN). But I think it's best to use your own Discord community if you have one. Else we can scrap that for now (i.e discard the community links in the document) until a community is set up
- Secondly, the sponsor button in the **README** points to Eddie's sponsorship page. So, I used the same URL under the **Financial contribution** section.

- I have highlighted them to facilitate modification if required

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+- You can append common abbreviations/acronyms used for communicating through the internet (social media) to the list of definitions in the abbreviations database.
+- Do not make duplicate entries (always confirm that an entry does not exist by searching through the JSON file.
+- Only English abbreviations are currently allowed.
+- Ensure that your entries are error-free by double-checking before staging your files.
+- Your entries should be in **T**itle **C**ase (e.g lgtm: **L**ooks **G**ood **To** **Me**, not looks good to me.)
+
+## Issues
+
+- Always create [issues](https://github.com/Njong392/Abbreve/issues) before making Pull Requests.
+- If you're not certain of the changes that you want to make, get in touch with us first!  You can [chat with us](Discord Link) or file an [issue](https://github.com/Njong392/Abbreve/issues).
```
Comment (slim-codes) [inline CONTRIBUTING.md:25]: Discord Link Suggestion

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+## Credits
+
+A big appreciation to all our wonderful contributors. You lot are awesome.
+
+[![Contributors](https://contrib.rocks/image?repo=Njong392/Abbreve)](https://github.com/Njong392/Abbreve/graphs/contributors)
+
+## Remarks
+
+If something is missing here, or you feel something is not well described, either directly open a PR or ask in one of the [Community platforms](Discord link or Discussions link if any).
```
Comment (slim-codes) [inline CONTRIBUTING.md:150]: Discord Link Suggestion

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  Provide a detailed description of the bug in the PR. Live demo preferred.
+
+  - Your PR should only do what its name implies (fixing a specific bug, implementing a specific feature, etc). Do not introduce unrelated changes like code refactors, spelling/grammar changes, etc. Otherwise, your PR will be rejected. A PR should focus on the fix and not code style, this makes it easier to trace changes in the future.
+  
+> **Note**: Always fill the PR template properly. Failing to do so will result in getting your PR rejected.
+
+## Financial contribution
+
+This is an Open Source project without corporate backing. Therefore we are open to any form of financial assistance towards supporting the maintainers' work and expenses.
+[Become a sponsor through the GitHub Sponsors program](https://github.com/sponsors/eddiejaoude)
```
Comment (slim-codes) [inline CONTRIBUTING.md:140]: Financial Contribution Section

**Review by slim-codes — COMMENTED** [review]
# Highlighted Issues

**Review by Njong392 — CHANGES_REQUESTED** [review]

**Review by adropofilm — APPROVED** [review]

**Pull request merged**